src/Github/Api/V3/SetDefaultBranchThroughApiCall.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Github\Api\V3;
+
+use Laminas\AutomaticReleases\Git\Value\BranchName;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Webmozart\Assert\Assert;
+
+use function Safe\json_decode;
+use function Safe\json_encode;
+
+final class SetDefaultBranchThroughApiCall implements SetDefaultBranch
+{
+    private const API_ROOT = 'https://api.github.com/';
+
+    private RequestFactoryInterface $messageFactory;
+
+    private ClientInterface $client;
+
+    private string $apiToken;
+
+    /** @psalm-param non-empty-string $apiToken */
+    public function __construct(
+        RequestFactoryInterface $messageFactory,
+        ClientInterface $client,
+        string $apiToken
+    ) {
+        $this->messageFactory = $messageFactory;
+        $this->client         = $client;
+        $this->apiToken       = $apiToken;
+    }
+
+    public function __invoke(
+        RepositoryName $repository,
+        BranchName $defaultBranch
+    ): void {
+        $request = $this->messageFactory
+            ->createRequest(
+                'PATCH',
```

Comment:
`\Fig\Http\Message\RequestMethodInterface::METHOD_PATCH`?

Reply:
Good one! I didn't even know of these constants \o/

Reply:
Hmm, no, those interfaces don't exist in current dependencies :\

Reply:
it's here https://github.com/php-fig/http-message-util/blob/master/src/RequestMethodInterface.php i.e. in `fig/http-message-util`

Reply:
Yeh, I would avoid importing a new package just for that